### DISCUSSION TITLE
Avoid newline character "\n" when indexing a PDF file

### DISCUSSION THREAD

**Discussion by pandveera** [dadoonet/fscrawler#1577] [Q&A] [answered]
When I try to create an elasticsearch index by reading a PDF file, the newline character "\n" is added in the index, is any solution to avoid it? and also is any solution to avoid the special character in the document.

**Comment by dadoonet** [comment, marked as answer]
The only way I can imagine today, is to use an [elasticsearch ingest pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/current/gsub-processor.html) which can transform the extracted text.

We could think of such a transformation in FSCrawler itself by implementing something similar to https://fscrawler.readthedocs.io/en/latest/admin/fs/local-fs.html#filter-content

Instead of ignoring the document, we can replace using a regular expression...

**Comment by pandveera** [comment]
Thanks for the reply.
